common/useFissionAuth.ts

```diff
@@ -0,0 +1,171 @@
+import { useEffect, useState } from 'react'
+
+import * as webnative from 'webnative'
+import * as webnativeFilecoin from 'webnative-filecoin';
+
+import * as C from '@common/constants';
+import * as R from '@common/requests';
+
+import Cookies from 'js-cookie';
+
+webnative.setup.debug({ enabled: true })
+
+export function useFissionAuth() {
+  const [state, setState] = useState<webnative.State>(null)
+  let fs;
+  let authScenario: webnative.Scenario | null = null;
+  let username: string = null;
+
+
+  /** Webnative Initialization
+   * Load webnative and configure permissions.
+   */
+
+  useEffect(() => {
+    async function getState() {
+      const result = await webnative.initialise({
+        permissions: {
+          // The Estuary token is stored in app storage
+          // at bgins/estuary-www.
+          app: {
+            name: 'estuary-www',
+            creator: 'bgins',
+          },
+          fs: {
+            // The cosigner key is stored in the private filesystem
+            // at the path Keychain/fil-cosigner.
+            private: [webnativeFilecoin.DEFAULT_KEY_PERMISSION]
+          }
+        },
+      })
+      setState(result)
+    }
+
+    getState()
+  }, [])
+
+
+  /** User and filesystem initialization
+   * If the user is authenticated with Fission, set their authScenario,
+   * filesytem, and username.
+   */
+
+  switch (state?.scenario) {
+    case webnative.Scenario.AuthSucceeded:
+    case webnative.Scenario.Continuation:
+      authScenario = state.scenario;
+      fs = state.fs;
+      username = state.username;
+      break;
+
+    default:
+      break;
+  }
+
+
+  /** Authorize
+   * Redirect the user to the Fission auth lobby where permission to use
+   * their filesystem will be requested. If they are new to Fission, they will
+   * first be asked to create an account.
+   */
+
+  const authorise = (redirectBackTo: string) => {
+    if (state) {
+      webnative.redirectToLobby(state.permissions, `http://localhost:4444/${redirectBackTo}`)
```

Comment:
We need the full URL to redirect back to when we return from the Fission Auth lobby. This is hardcoded for the moment, but setting it as an environment variable or something else would be better,

Reply:
Yeah makes sense we can just check process.env.NODE_ENV and pass it into the constructor of this utility method.

Reply:
I've pushed a change to get the `host` and `protocol` from the server side props. Ultimately, it would be nice to move away from server reliance, but this should work with the current implementation.

Getting the protocol doesn't seem to be all that consistent in node, so this is implemented by checking the `host` and setting the protocol to `http` for `localhost` and `https` for anything else. But -- I am not the most experienced node developer, and I am open to alternatives here.